app/src/features/surveys/view/SurveyAnimals.tsx

```diff
@@ -70,13 +85,47 @@ const SurveyAnimals: React.FC = () => {
   };
 
   const DeviceFormValues: IAnimalTelemetryDevice = {
-    device_id: 0,
+    device_id: '' as unknown as number,
     device_make: '',
-    frequency: 0,
+    frequency: '' as unknown as number,
     frequency_unit: '',
     device_model: '',
-    attachment_start: '',
-    attachment_end: undefined
+    deployments: [
+      {
+        deployment_id: '',
+        attachment_start: '',
+        attachment_end: undefined
+      }
+    ]
+  };
+
+  const obtainDeviceFormInitialValues = (mode: TelemetryDeviceFormMode) => {
+    switch (mode) {
+      case 'add':
+        return [DeviceFormValues];
+      case 'edit': {
+        const deployments = deploymentData?.filter((a) => a.critter_id === currentCritterbaseCritterId);
+        if (deployments) {
+          //Any suggestions on something better than this reduce is welcome.
+          //Idea is to transform flat rows of {device_id, ..., deployment_id, attachment_end, attachment_start}
+          //to {device_id, ..., deployments: [{deployment_id, attachment_start, attachment_end}]}
+          const red = deployments.reduce((acc, curr) => {
+            const currObj = acc.find((a: any) => a.device_id === curr.device_id);
+            const { attachment_end, attachment_start, deployment_id, ...rest } = curr;
+            const deployment = { deployment_id, attachment_start, attachment_end };
+            if (!currObj) {
+              acc.push({ ...rest, deployments: [deployment] });
+            } else {
+              currObj.deployments?.push(deployment);
+            }
+            return acc;
+          }, [] as IAnimalTelemetryDevice[]);
```

Comment:
You can avoid casting if you just give this type to `acc` at the top of the closure